drivers/si7021/si7021.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2016 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_si7021
+ * @{
+ *
+ * @file
+ * @brief       Implementation of SI7021 sensor driver.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+
+#include "si7021.h"
+
+#include "xtimer.h"
+
+int si7021_test(si7021_t *dev)
+{
+    uint8_t revision = si7021_get_revision(dev);
+
+    if (revision != SI7021_REVISION_1 && revision != SI7021_REVISION_2) {
+        return -1;
+    }
+
+    uint8_t id = (si7021_get_id(dev) >> 24) & 0xff;
+
+    if (id != SI7021_ID) {
+        return -2;
+    }
+
+    return 0;
+}
+
+int si7021_init(si7021_t *dev, i2c_t i2c_dev)
+{
+    dev->i2c_dev = i2c_dev;
+
+    /* setup the i2c bus */
+    i2c_acquire(dev->i2c_dev);
+    int result = i2c_init_master(dev->i2c_dev, I2C_SPEED_NORMAL);
+
+    if (result != 0) {
+        i2c_release(dev->i2c_dev);
+        return result;
+    }
+    i2c_release(dev->i2c_dev);
+
+    /* initialize the peripheral */
+    si7021_reset(dev);
+
+    return 0;
+}
+
+void si7021_reset(si7021_t *dev)
```

Comment:
I would make this function as part of the `init()`. This saves some `acquire/release` calls